lldb-eval/eval.cc

```diff
@@ -981,6 +981,11 @@ Value Interpreter::EvaluateComparison(BinaryOpKind kind, Value lhs, Value rhs) {
   }
 
   // Must be pointer/integer and/or nullptr comparison.
+  if (target_.GetAddressByteSize() == 4) {
+    bool ret = Compare(kind, static_cast<uint32_t>(lhs.GetUInt64()),
+                       static_cast<uint32_t>(rhs.GetUInt64()));
+    return CreateValueFromBool(target_, ret);
+  }
```

Comment:
Hm, does `GetUInt64()` return incorrect values for 4-byte pointers (i.e. with garbage in the other 4 bytes)? I think this is not the only place where we use `GetUInt64()` to get the "pointer value" from the variable, it should probably be handled in other places as well.